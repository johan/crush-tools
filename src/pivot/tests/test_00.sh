test_number=00
description="minimal"

input=$test_dir/test_$test_number.in
expected=$test_dir/test_$test_number.expected

subtest=1
output=$test_dir/test_$test_number.$subtest.output
$bin -k -f 1 -p 2 -v 3 $input > $output
if [ $? -ne 0 ] || [ "`diff -q $output $expected`" ]; then
  test_status $test_number $subtest "$description (indexes)" FAIL
else
  test_status $test_number $subtest "$description (indexes)" PASS
  rm $output
fi

subtest=2
output=$test_dir/test_$test_number.$subtest.output
$bin -F Field-One -P Field-Two -A File-A-Field $input > $output
if [ $? -ne 0 ] || [ "`diff -q $output $expected`" ]; then
  test_status $test_number $subtest "$description (labels)" FAIL
else
  test_status $test_number $subtest "$description (labels)" PASS
  rm $output
fi
