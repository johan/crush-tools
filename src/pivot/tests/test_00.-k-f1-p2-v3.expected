Field-One	Two Key 3: File-A-Field	Two Key 4: File-A-Field
One Key 1	10	100
One Key 2	1000	10000
